blog/2023-02-09-http-s3-blog/index.md

```diff
@@ -0,0 +1,220 @@
+vanus-controller-1               1/1     Running   0          72s
+vanus-controller-2               1/1     Running   0          69s
+vanus-gateway-8677fc868f-rmjt9   1/1     Running   0          97s
+vanus-store-0                    1/1     Running   0          96s
+vanus-store-1                    1/1     Running   0          68s
+vanus-store-2                    1/1     Running   0          68s
+vanus-timer-5cd59c5bf-hmprp      1/1     Running   0          97s
+vanus-timer-5cd59c5bf-pqkd5      1/1     Running   0          97s
+vanus-trigger-7685d6cc69-8jgsl   1/1     Running   0          97s
+```
+
+- Install vsctl (the command line tool).
+
+```shell
+curl -O https://download.linkall.com/vsctl/latest/linux-amd64/vsctl
+chmod ug+x vsctl
+mv vsctl /usr/local/bin
+```
+
+- Set the endpoint for vsctl.
+
+```shell
+export VANUS_GATEWAY=192.168.49.2:30001
+```
+
+- Create an Eventbus to store your events.
+
+```shell
+vsctl eventbus create --name http-s3
++----------------+------------------+
+|     RESULT     |      EVENTBUS    |
++----------------+------------------+
+| Create Success | http-s3          |
++----------------+------------------+
+```
+
+### Step 2: Make directory for S3 Sink Connector and Create Config file
+
+- Make S3 Sink Directory
+
+```shell
+mkdir s3-sink
+```
+
+- Create Config file
+
+```shell
+cat << EOF > config.yml
+port: 8080
+aws:
+  access_key_id: your_access_key
+  secret_access_key: your_secret_key
+region: "your_region"
+bucket: "your_bucket_name"
+scheduled_interval: 10
+EOF
+```
+
+- Use**docker run** to run the S3 sink config.yml file
+
+```shell
+docker run -it --rm \
+  -p 8082:8080 \
+  -v ${PWD}:/vanus-connect/config \
+  --name sink-aws-s3 public.ecr.aws/vanus/connector/sink-aws-s3 &
+```
+
+**Note: I ran this in the Background of my terminal, if you wish to see the outputs, remove the ampersand (&) at the end**
+
+### Step 3: Make directory for HTTP Source Connector and Create Config file
+
+- Make HTTP Source Directory
+
+```shell
+mkdir http-source
+```
+
+- Create Config file
+
+```shell
+cat << EOF > config.yml
+target: http://192.168.49.2:30002/gateway/http-to-s3
+port: 31081
+EOF
+```
+
+- Use **docker run** to run the HTTP Source config.yml file
+
+```shell
+docker run -it --rm --network=host \
+  -v ${PWD}:/vanus-connect/config \
+  --name source-http public.ecr.aws/vanus/connector/source-http &
+```
+
+**Note: I ran this in the Background of my terminal, if you wish to see the outputs, remove the ampersand (&) at the end**
+
+### Step 4: Create Subscription
+
+- The Subscription is a relationship established between a Sink and an Eventbus. The Subscription reflects the Sink's interest in receiving events and describes the method for how to deliver those events. To create a subcription, use
+
+```shell
+vsctl subscription create --name http \
+  --eventbus http-to-s3 \
+  --sink 'http://ip10-1-39-4-cecpi79ajm80o97dfdug-8082.direct.play.linkall.com'
```

Comment:
The link is different for each user, so it wouldn't work this way you have to let them get there special link